aiida/engine/daemon/execmanager.py

```diff
@@ -139,8 +139,12 @@ def upload_calculation(node, transport, calc_info, folder, inputs=None, dry_run=
     for code in input_codes:
         if code.is_local():
             # Note: this will possibly overwrite files
-            for f in code.get_folder_list():
-                transport.put(code.get_abs_path(f), f)
+            for f in code.list_object_names():
+                with NamedTemporaryFile(mode='wb+') as handle:
+                    handle.write(code.get_object_content(f, mode='rb'))
+                    handle.flush()
+                    handle.seek(0)
+                    transport.put(handle.name, f)
```

Comment:
This might be because I'm still not familiar with `transport` and `code` objects, but from the current commit description (which only mentions that `get_folder_list` no longer exists) I don't understand why the change in the parameters of `transport.put(...)`. It seems to change the loop over folders for a loop over objects (file names, correct?), but then I don't understand why you make a temporary file and pass the name of that instead of something related to the looping objects. Also, I don't understand the process of writing content, flushing and rewinding said temporary file.

Again, I'm not sure if this merits any code comment, but maybe just a brief explanation for this changes in the commit message?

Reply:
That change is because the new API doesn't expose directly an absolute path to the file to be put on the remote computer (which wouldn't be possible on an object store backend). Since `transport.put` wants a filename, the content is first copied to a temporary file.

This part was clear to me, but I have been doing a fair bit of migrating between these two.. so maybe the commit could indeed be more explicit about that.

Reply:
I will include the reasoning for this and a reference to #2579 which will allow this hack to be removed.